zerver/lib/message.py

```diff
@@ -1188,22 +1188,29 @@ def check_user_group_mention_allowed(sender: UserProfile, user_group_ids: List[i
     sender_is_system_bot = is_cross_realm_bot_email(sender.delivery_email)
 
     for group in user_groups:
-        can_mention_group = group.can_mention_groups.all()[0]
+        can_mention_groups = group.can_mention_groups.all()
+        can_mention_group_names = {group.name for group in can_mention_groups}
+        if SystemGroups.EVERYONE in can_mention_group_names:
+            continue
 
         if sender_is_system_bot:
-            if can_mention_group.name == SystemGroups.EVERYONE:
-                continue
             raise JsonableError(
                 _(
-                    "You are not allowed to mention user group '{user_group_name}'. You must be a member of '{can_mention_group_name}' to mention this group."
-                ).format(user_group_name=group.name, can_mention_group_name=can_mention_group.name)
+                    "You are not allowed to mention user group '{user_group_name}'. You must be a member of any one of these groups to mention this group - {can_mention_group_string}."
```

Comment:
It might be better to just drop the list of required groups from the error message and just stop at the first sentence.